packages/nimble-components/src/select/specs/option-groups-hld.md

```diff
@@ -0,0 +1,106 @@
+# Select and Combobox option groups
+
+## Problem Statement
+
+It is a common request to categorize the options that are present in the dropdown for a `Select`/`Combobox` component. By supporting categories, developers are able to logically group options, and help end-users to quickly find the relevant option. This will also support use cases that need the same display values under separate categories.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Interaction Design](./IxD.md#groups)
+-   [Sectioned Combobox (issue 791)](https://github.com/ni/nimble/issues/791)
+-   [Prototype branch Storybook](https://60e89457a987cf003efc0a5b-cvmxlqwglb.chromatic.com/?path=/docs/components-select--docs) ([code](https://github.com/ni/nimble/tree/groupable-select))
+
+## Implementation / Design
+
+The HTML native `select` supports option groups with the native `optgroup` element. Configuring the `select` to use the `optgroup` takes the following form:
+
+```html
+<select name="cars">
+    <optgroup label="Swedish Cars">
+        <option value="volvo">Volvo</option>
+        <option value="saab">Saab</option>
+    </optgroup>
+    <optgroup label="German Cars">
+        <option value="mercedes">Mercedes</option>
+        <option value="audi">Audi</option>
+    </optgroup>
+</select>
+```
+
+The Nimble `Select` (and `Combobox`) will follow the same approach, and provide a new custom element, `nimble-list-option-group`, that mirrors the `optgroup` behavior/API:
+
+```html
+<nimble-select name="cars">
+    <nimble-list-option-group label="Swedish Cars">
+        <nimble-list-option value="volvo">Volvo</option>
+        <nimble-list-option value="saab">Saab</option>
+    </nimble-list-option-group>
+    <nimble-list-option-group label="German Cars">
+        <nimble-list-option value="mercedes">Mercedes</option>
+        <nimble-list-option value="audi">Audi</option>
+    </nimble-list-option-group>
+</nimble-select>
+```
+
+#### Visual separators
+
+The visual design calls for option groups to have a visual separator between them:
+
+![](./spec-images/option-group-separator.png)
+
+To accomplish this, a client will be required to insert an `<hr>` element between each group. This mirrors the API for the `Menu`.
+
+#### Top-level Options
+
+While not a common use-case, clients can provide `nimble-list-option`s as siblings to `nimble-list-option-group`s. In such cases, the client will be expected to insert the appropriate `<hr>` element between a `nimble-list-option-group` and set of `nimble-list-option`s that immediately follow it.
+
+### API
+
+The `ListOptionGroup` class API should be fairly lightweight:
+
+```ts
+public class ListOptionGroup {
+    @attr
+    public label?: string;
+}
+```
+
+In addition to the `label` attribute, the `ListOptionGroup` will also support slotted labels. This will result in a configuration that looks like the following:
```

Comment:
What's the rationale for supporting multiple APIs to provide a label? If we allow the slotted approach do we need to document / enforce restrictions?
1. can the slotted label be in any order or must it be before all options?
2. can you slot multiple labels?
3. can you slot a span containing a label? (This would likely be important to apply a targeted `i18n` attribute in Angular)
4. can you slot more complex HTML content like an icon plus a label?

My initial inclination is to avoid all these questions and only support the label attribute, but that may change when I hear the rationale for the slotted API.

Reply:
The slotted API approach was suggested because it was more consistent with other Nimble component APIs for how such things are configured (like a Button's text). The label attribute seemed like a snowflake API, despite its consistency with the native `optgroup` API. The multiple API approach was suggested by @rajsite I believe primarily to promote consistency with native APIs (which could be useful for those moving away from native components to our own) as well as acknowledging that to be consistent with other Nimble APIs would lean heavily towards a slotted approach.

Regarding your specific questions about what kind of restrictions we would have, I would expect we would be as lax with them as we are for the default slotted content in a button.

Admittedly, I don't feel too strongly with the approach we decide to take, and can be swayed easily (my original thought was just to do the `label` attribute).

Reply:
We spoke offline and agreed:
1. keep both the attribute and default slot APIs
2. attribute would win over the slot
3. we'd document the common cases (single label before all the options, plus a span containing a label) and all the other cases above would be considered unsupported for now. They could become supported if people ask for them and we document them.